R/readEurObservER.R

```diff
@@ -44,7 +44,81 @@ readEurObservER <- function(subtype) {
+        reportPeriod <- gsub(".*(\\d{4}).*", "\\1", file)
+        wideTable <- read.csv(file, header = FALSE)
+
+        if (reportPeriod == "2013") {
+          years <- switch(gsub(".*Table|.csv", "", file), `5` = 2011, `6` = 2012)
+          wideTable <- wideTable[, !grepl("[rR]enewable heat", wideTable[1, ])]
+        } else {
+          years <- grep("\\d{4}", wideTable[1, ], value = TRUE)
+        }
+
+        nDataCols <- (ncol(wideTable) - 1) / length(years)
+
+        longTable <- do.call("rbind", lapply(seq_along(years), function(i) {
+          subTable <- wideTable[ifelse(reportPeriod == "2013", 1, 2):nrow(wideTable),
+                                c(1, ((i - 1) * nDataCols + 2):(i * nDataCols + 1))]
+          colnames(subTable) <- subTable[1, ]
+          subTable <- subTable[-1, ]
+          subTable[["period"]] <- years[[i]]
+          return(subTable)
+        }))
+
+        colnames(longTable)[1] <- "region"
+        longTable <- longTable %>%
+          gather("variable", "value", -any_of(c("region", "period"))) %>%
+          mutate(reportPeriod = reportPeriod)
+
+        return(longTable)
+      }))
+
+      # tidy data
+      variableMatching <- c(
+        `ASHP`                    = "ASHP",
+        `Aerothermal HP`          = "ASHP",
+        `Aerothermal heat pumps`  = "ASHP",
+        `includ. air- water HP`   = "ASHP_airWater",
+        `GSHP`                    = "GSHP",
+        `Ground source heat pump` = "GSHP",
+        `Geothermal HP`           = "GSHP",
+        `Geothermal heat pumps`   = "GSHP",
+        `Total PAC`               = "Total",
+        `Total heat pumps`        = "Total",
+        `Total HP in operation`   = "Total"
+      )
+      data <- data %>%
+        mutate(value = ifelse(.data[["value"]] == "-", NA, .data[["value"]]),
+               value = as.numeric(gsub("o|O", "0", gsub(" ", "", .data[["value"]]))),
+               variable = variableMatching[.data[["variable"]]],
+               period = as.numeric(.data[["period"]]),
+               reportPeriod = as.numeric(.data[["reportPeriod"]]),
+               region = iconv(.data[["region"]], "latin1", "ASCII", sub = ""),
+               region = gsub("\\*", "", .data[["region"]]),
+               region = gsub("Czechia", "Czech Republic", .data[["region"]]),
+               region = gsub("^Netherland$", "Netherlands", .data[["region"]]),
+               region = gsub("United(-K| k)ingdom", "United Kingdom", .data[["region"]]),
+               region = gsub("Lituania", "Lithuania", .data[["region"]]),
+               region = gsub("Hungarie", "Hungary", .data[["region"]]))
+      sumIsWrong <- data %>%
+        group_by(across(all_of(c("region", "period", "reportPeriod")))) %>%
+        filter(abs(.data[["value"]][.data[["variable"]] == "Total"] -
+                     .data[["value"]][.data[["variable"]] == "ASHP"] -
+                     .data[["value"]][.data[["variable"]] == "GSHP"]) > 1) %>%
```

Comment:
Is it intentional that you do not reflect "ASHP_airWater" here? And I'm guessing that you set the threshold to 1 as a sufficiently small deviation that can still be tolerated?

Reply:
Yes, it is intentional. `ASHP` should include `ASHP_airWater`, so we would double count.
![grafik](https://github.com/user-attachments/assets/aedade7c-4f50-402f-8fa6-ad2b9056281c)
Not sure why I had a 1 there. Maybe I wanted to use `>=`. Anyways, I replaced it with 0 and tested that the function still runs through. Thanks for spotting.